src/communication/channelmanager.cpp

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <aos/common/tools/timer.hpp>
+
+#include "channel.hpp"
+#include "channelmanager.hpp"
+#include "log.hpp"
+
+namespace aos::zephyr::communication {
+
+Error ChannelManager::Init(TransportItf& transport)
+{
+    LOG_INF() << "Init channel manager";
+
+    mTransport = &transport;
+
+    return Run();
+}
+
+RetWithError<ChannelItf*> ChannelManager::CreateChannel(uint32_t port)
+{
+    UniqueLock lock(mMutex);
+
+    LOG_DBG() << "Create channel port=" << port;
+
+    ChannelInfo info = {port, info.mChannel = aos::MakeShared<Channel>(&mChanAllocator, mTransport, port)};
+
+    if (auto err = mChannels.PushBack(info); !err.IsNone()) {
+        return {nullptr, err};
+    }
+
+    mCondVar.NotifyAll();
```

Comment:
it's not clear why cond var required.

Reply:
```
Error ChannelManager::HandleRead()
{
    while (true) {
        {
            aos::UniqueLock lock(mMutex);

            mCondVar.Wait(lock, [this] { return !mChannels.IsEmpty() || mClose; });

            if (mClose) {
                return ErrorEnum::eNone;
            }
        }
```

before reading something from transport and passing this data to the channel we need to wait for at least one channel.